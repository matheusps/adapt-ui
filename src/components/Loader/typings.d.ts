type Measure = 'sm' | 'md' | 'lg' | 'xl'
type Speed = 'fast' | 'slow' | 'normal'
type Color = 'primary'
interface LoaderProps {
  color?: Color
  speed?: Speed
  gap?: Measure
  thickness?: Measure
  size?: string
}
