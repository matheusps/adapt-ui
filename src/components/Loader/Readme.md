Loader colors

```js
<Loader />
<Loader color='secondary' />
<Loader color='success' />
```
