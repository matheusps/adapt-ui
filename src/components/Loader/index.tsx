import React, { FunctionComponent } from 'react'

import AnimatedSvg from './helpers/AnimatedSvg'
import { getGap, getSpeed, getThickness } from './helpers/helpers'

const Loader: FunctionComponent<LoaderProps> = ({
  thickness,
  size,
  speed,
  color,
  gap,
}) => (
  <AnimatedSvg
    height={size}
    width={size}
    style={{ animationDuration: `${getSpeed(speed!)}ms` }}
    role="img"
    viewBox="0 0 32 32"
  >
    <circle
      role="presentation"
      cx={16}
      cy={16}
      r={14 - getThickness(thickness!) / 2}
      stroke={color}
      fill="none"
      strokeWidth={getThickness(thickness!)}
      strokeDasharray={Math.PI * 2 * (11 - getGap(gap!))}
      strokeLinecap="round"
    />
  </AnimatedSvg>
)

Loader.defaultProps = {
  color: 'rgba(0,0,0,0.4)',
  size: '2rem',
  thickness: 'md',
  speed: 'normal',
  gap: 'lg',
}

export default Loader
