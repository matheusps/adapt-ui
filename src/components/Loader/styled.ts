import styled, { keyframes } from 'styled-components'
import { get } from 'lodash'

const infiniteSpin = keyframes`
  from {transform: rotate(0deg)}
  to {transform: rotate(360deg)}
`

const Svg = styled.svg`
  transition-property: transform;
  animation-name: ${infiniteSpin};
  animation-iteration-count: infinite;
  animation-timing-function: linear;
`

const Circle = styled.circle<withGoal>`
  stroke: ${props => get(props.theme.colors.goal, props.goal)};
`

export { Svg, Circle }
