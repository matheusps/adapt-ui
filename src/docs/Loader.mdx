---
## name: Loader
---

import { Playground } from 'docz'
import Loader from '../components/Loader'

# Loader

## Goals

<Playground>
  <Loader goal="create" />
  <Loader goal="destroy" />
  <Loader goal="info" />
  <Loader goal="confirm" />
  <Loader goal="warn" />
</Playground>

## Size

<Playground>
  <Loader size="sm" />
  <Loader size="md" />
  <Loader size="lg" />
  <Loader size="xl" />
</Playground>

## Thickness

<Playground>
  <Loader thickness="sm" />
  <Loader thickness="md" />
  <Loader thickness="lg" />
  <Loader thickness="xl" />
</Playground>

## Gap

<Playground>
  <Loader gap="sm" />
  <Loader gap="md" />
  <Loader gap="lg" />
  <Loader gap="xl" />
</Playground>

## Speed

<Playground>
  <Loader speed="slow" />
  <Loader speed="normal" />
  <Loader speed="fast" />
</Playground>
