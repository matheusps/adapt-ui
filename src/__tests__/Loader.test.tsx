import React from 'react'
import { render } from 'react-testing-library'
import Loader from '../components/Loader'

describe('Container component', () => {
  it('should render', () => {
    const component = render(<Loader />)
    expect(component).toBeTruthy()
  })

  it('should match snapshot as default', () => {
    const component = render(<Loader />)
    expect(component.asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with all props', () => {
    const component = render(
      <Loader gap="xl" size="3rem" color="black" speed="fast" thickness="md" />
    )
    expect(component.asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with all measures', () => {
    const component = render(
      <div>
        <Loader thickness="sm" gap="sm" />
        <Loader thickness="md" gap="md" />
        <Loader thickness="lg" gap="lg" />
        <Loader thickness="xl" gap="xl" />
      </div>
    )
    expect(component.asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with all speeds', () => {
    const component = render(
      <div>
        <Loader speed="slow" />
        <Loader speed="normal" />
        <Loader speed="fast" />
      </div>
    )
    expect(component.asFragment()).toMatchSnapshot()
  })
})
