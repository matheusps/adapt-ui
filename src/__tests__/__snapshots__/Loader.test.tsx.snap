// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should match snapshot as default 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-animation-name: MQNGw;
  animation-name: MQNGw;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c1 {
  stroke: #020100;
}

<svg
    class="c0"
    height="2rem"
    role="img"
    style="animation-duration: 750ms;"
    viewBox="0 0 32 32"
    width="2rem"
  >
    <circle
      class="c1"
      cx="16"
      cy="16"
      fill="none"
      r="12"
      role="presentation"
      stroke-dasharray="43.982297150257104"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;

exports[`Container component should match snapshot with all measures 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-animation-name: MQNGw;
  animation-name: MQNGw;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c1 {
  stroke: #020100;
}

<svg
    class="c0"
    height="sm"
    role="img"
    style="animation-duration: 750ms;"
    viewBox="0 0 32 32"
    width="sm"
  >
    <circle
      class="c1"
      cx="16"
      cy="16"
      fill="none"
      r="12"
      role="presentation"
      stroke-dasharray="43.982297150257104"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;

exports[`Container component should match snapshot with all measures 2`] = `
<DocumentFragment>
  .c0 {
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-animation-name: MQNGw;
  animation-name: MQNGw;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c1 {
  stroke: #020100;
}

<svg
    class="c0"
    height="md"
    role="img"
    style="animation-duration: 750ms;"
    viewBox="0 0 32 32"
    width="md"
  >
    <circle
      class="c1"
      cx="16"
      cy="16"
      fill="none"
      r="12"
      role="presentation"
      stroke-dasharray="43.982297150257104"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;

exports[`Container component should match snapshot with all measures 3`] = `
<DocumentFragment>
  .c0 {
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-animation-name: MQNGw;
  animation-name: MQNGw;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c1 {
  stroke: #020100;
}

<svg
    class="c0"
    height="lg"
    role="img"
    style="animation-duration: 750ms;"
    viewBox="0 0 32 32"
    width="lg"
  >
    <circle
      class="c1"
      cx="16"
      cy="16"
      fill="none"
      r="12"
      role="presentation"
      stroke-dasharray="43.982297150257104"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;

exports[`Container component should match snapshot with all measures 4`] = `
<DocumentFragment>
  .c0 {
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-animation-name: MQNGw;
  animation-name: MQNGw;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c1 {
  stroke: #020100;
}

<svg
    class="c0"
    height="xl"
    role="img"
    style="animation-duration: 750ms;"
    viewBox="0 0 32 32"
    width="xl"
  >
    <circle
      class="c1"
      cx="16"
      cy="16"
      fill="none"
      r="12"
      role="presentation"
      stroke-dasharray="43.982297150257104"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;

exports[`Container component should match snapshot with all props 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-animation-name: MQNGw;
  animation-name: MQNGw;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c1 {
  stroke: #020100;
}

<svg
    class="c0"
    height="3rem"
    role="img"
    style="animation-duration: 600ms;"
    viewBox="0 0 32 32"
    width="3rem"
  >
    <circle
      class="c1"
      cx="16"
      cy="16"
      fill="none"
      r="12"
      role="presentation"
      stroke-dasharray="37.69911184307752"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;

exports[`Container component should match snapshot with all speeds 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-animation-name: MQNGw;
  animation-name: MQNGw;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c1 {
  stroke: #020100;
}

<svg
    class="c0"
    height="2rem"
    role="img"
    style="animation-duration: 900ms;"
    viewBox="0 0 32 32"
    width="2rem"
  >
    <circle
      class="c1"
      cx="16"
      cy="16"
      fill="none"
      r="12"
      role="presentation"
      stroke-dasharray="43.982297150257104"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;

exports[`Container component should match snapshot with all speeds 2`] = `
<DocumentFragment>
  .c0 {
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-animation-name: MQNGw;
  animation-name: MQNGw;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c1 {
  stroke: #020100;
}

<svg
    class="c0"
    height="2rem"
    role="img"
    style="animation-duration: 750ms;"
    viewBox="0 0 32 32"
    width="2rem"
  >
    <circle
      class="c1"
      cx="16"
      cy="16"
      fill="none"
      r="12"
      role="presentation"
      stroke-dasharray="43.982297150257104"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;

exports[`Container component should match snapshot with all speeds 3`] = `
<DocumentFragment>
  .c0 {
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-animation-name: MQNGw;
  animation-name: MQNGw;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c1 {
  stroke: #020100;
}

<svg
    class="c0"
    height="2rem"
    role="img"
    style="animation-duration: 600ms;"
    viewBox="0 0 32 32"
    width="2rem"
  >
    <circle
      class="c1"
      cx="16"
      cy="16"
      fill="none"
      r="12"
      role="presentation"
      stroke-dasharray="43.982297150257104"
      stroke-linecap="round"
      stroke-width="4"
    />
  </svg>
</DocumentFragment>
`;
